website/source/docs/guides/production-acls.html.md

```diff
@@ -0,0 +1,484 @@
+for production workloads. 
+
+After [bootstrapping the ACL
+system](/advanced/day-1-operations/production-acls#bootstrap-the-acl-system),
+you will learn how to create tokens with minimum privileges for:
+
+* [Servers and Clients](/advanced/day-1-operations/production-acls#appy-individual-tokens-to-agents)
+* [Services](/advanced/day-1-operations/production-acls#apply-individual-tokens-to-services)
+* [DNS](/advanced/day-1-operations/production-acls#token-for-dns) 
+*  [Consul KV](/advanced/day-1-operations/production-acls#consul-kv-tokens) 
+* [Consul UI](/advanced/day-1-operations/production-acls#consul-ui-tokens) 
+
+~> **Important:** For best results, use this guide during the [initial
+deployment](/advanced/day-1-operations/deployment-guide) of a Consul (version
+1.4.0 or newer) datacenter. Specifically, you should have already installed all
+agents and configured initial service definitions, but you should not yet rely
+on Consul for any service discovery or service configuration operations.  
+
+## Bootstrap the ACL System
+
+You will bootstrap the ACL system in two steps, enable ACLs and create the
+bootstrap token.  
+
+### Enable ACLs on the Agents
+
+To enable ACLs, add the following [ACL
+parameters](https://www.consul.io/docs/agent/options.html#configuration-key-reference)
+to the agent's configuration file and then restart the Consul service. If you
+want to reduce Consul client restarts, you can enable the ACLs 
+on them when you apply the token. 
+
+```
+# agent.hcl 
+{ 
+  acl = { 
+  	enabled = true, 
+  	default_policy = "deny" 
+  	} 
+} 
+```
+
+In this example, you configured the default policy of "deny", which means you
+are in whitelist mode. 
+
+~> Note: If you are bootstrapping ACLs on an existing datacenter, enable the
+ACLs on the agents first with `default_policy=allow`. Default policy allow will
+enable ACLs, but will allow all operations, allowing the cluster to function
+normally while you create the tokens and apply them. This will reduce downtime.
+You should update the configuration files on all the servers first and then
+initiate a rolling restart. 
+
+### Create the Initial Bootstrap Token
+
+To create the initial bootstrap token, use the `acl bootstrap` command on one
+of the servers. 
+
+``` 
+$ consul acl bootstrap 
+```
+
+The output gives you important information about the token, including the
+associated policy `global-management` and `SecretID`. 
+
+~> Note: By default, Consul assigns the `global-management` policy to the
+bootstrap token, which has unrestricted privileges. It is important to have one
+token with unrestricted privileges in case of emergencies; however you should
+only give a small number of administrators access to it. The `SecretID` is a
+UUID that you will use to identify the token when using the Consul CLI or HTTP
+API. 
+
+While you are setting up the ACL system, set the `CONSUL_HTTP_TOKEN`
+environment variable to the bootstrap token on one server, for example
```

Comment:
It does cut down on the branching factor of the guide to just pick one way of doing things and carry that through to the end instead of mentioning 2-3 different ways to achieve each step.

Reply:
I would say that we don't need to mention where you do it from.

Reply:
I would really like this to be another guide or part of the docs so we can just link out. It's complicated enough that I don't want to overwhelm users setting up ACLs since its already a complicated process. I'm going to create an Asana task for this so we can work on it soon.